javascript/test/memory.spec.js

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2016 OpenMarket Ltd
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+"use strict";
+
+var Olm = require('../olm');
+
+if (!Object.keys) {
+    Object.keys = function(o) {
+        var k=[], p;
+        for (p in o) if (Object.prototype.hasOwnProperty.call(o,p)) k.push(p);
+        return k;
+    }
+}
+
+describe("olm memory", function() {
+    beforeEach(function(done) {
+        Olm.init().then(function() {
+            done();
+        });
+    });
```

Comment:
Personally I prefer `async` syntax:

```js
beforeEach(async function(done) {
  await Olm.init();
  done();
});
```

Reply:
Me too, but I decided for this syntax for two reasons:
1. To keep it consistent with other tests in this codebase
2. Looks like Olm has test configurations that don't like `async` syntax, after changes I am seeing:
```
An asynchronous function was defined with the async keyword but also took a done callback. This is not supported and will stop working in the future
```

Reply:
Re. 2 - If you go with async, the function should not include the callback argument
```suggestion
    beforeEach(async function() {
      await Olm.init();
    });
```

Reply:
Oh interesting, unresolved conversation since that seems like a good idea